swifter/swifter.py

```diff
@@ -179,7 +179,6 @@ def resample(
         label=None,
         convention="start",
         kind=None,
-        loffset=None,
```

Comment:
I'm not really familiar enough with swifter's (or pandas') API, however assuming that this parameter is being removed from a public method then unfortunately I think this constitutes a breaking change for anyone using it.
I realise that that may be a small group of users given the age of `pandas < 1.1` (that being when this was deprecated), however if you're hoping to stick to SemVer then I think this change deserves a major version bump.

Reply:
This has got me thinking that it might be a better move to implement this change in a backwards-compatible manner.  That is, I could check what version the user is on and only use `loffset` if they are using `pandas < 2.0` (i think, as @bnavigator mentioned below, that it was actually removed in 2.0, but was the deprecation was introduced in 1.1)... though i could use more clarity on what the implies for `swifter`

Reply:
Proposed solution to avoid making a breaking change https://github.com/jmcarpenter2/swifter/pull/220/commits/debf0ae9b44f4e89c673f530b8ed01c162e8c79c